### PR TITLE
rm: factory definition: Describe a list of variables

### PR DESCRIPTION
**Pull request by angolini**
## Readiness

* [x] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview
Describe some variables and provide some references in a central place (the page where you see how to change the factory config file).

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 6a9aeee08e2f3123383c10850052806031667afb are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2341/docs/artifacts/html/index.html

File: source/reference-manual/factory/factory-definition.rst [hunk 1]
```diff
@@ -169,11 +174,42 @@ lmp:
...
+* **DISTRO**:
+               Define the distro being used. Reference :ref:`ref-linux-distro`.
+* **TUF_TARGETS_EXPIRE**:
+               Is used to change the default target expiration date (default 1y).
+* **SSTATE_CACHE_MIRROR**:
+               It defaults to the directory mounted on the SDK build container
+               and if this directory exists it is used as the source for the
+               shared state cache (sstate-cache) mirror. When the directory does
+               not exist the ``lmp-manifest`` value is used and should point to
+               the public HTTP shared state cache.
```
Comment (quaresmajose) [inline source/reference-manual/factory/factory-definition.rst:204]: -value is used and should point to the public HTTP shared state cache.
+value is used (currently it point to the public HTTP shared state cache).

File: source/reference-manual/factory/factory-definition.rst [hunk 2]
```diff
@@ -169,11 +174,42 @@ lmp:
...
 
+Variables
+^^^^^^^^^
+
+* **DEV_MODE**:
+               This is a flexible variable used to configure the source code
+               into development mode. The development mode should be defined
+               by the user. Reference: :ref:`ref-dev-mode`.
+* **BUILD_SDK**:
+               With this variable set to ``1`` the SDK artifact will be part
```
Comment (kprosise) [inline source/reference-manual/factory/factory-definition.rst:185]:
```suggestion
               With this variable set to ``1``, the SDK artifact will be part
```

File: source/reference-manual/factory/factory-definition.rst [hunk 3]
```diff
@@ -169,11 +174,42 @@ lmp:
...
+               of the build. Reference: :ref:`ref-building-sdk`.
+* **DISABLE_GPLV3**:
+               When set to ``1``, this variable configures the source code
+               to avoid the LmP default packages under GPLv3.
+               Reference: :ref:`ref-remove-gplv3`.
+* **ENABLE_PTEST**:
+               When set to ``1`` this variable creates an artifact based on the
```
Comment (kprosise) [inline source/reference-manual/factory/factory-definition.rst:192]:
```suggestion
               When set to ``1``, this variable creates an artifact based on the
```

File: source/reference-manual/factory/factory-definition.rst [hunk 4]
```diff
@@ -169,11 +174,42 @@ lmp:
...
+               image being built with the pattern ``<main tag>-ptest``. This
+               image includes the available Ptest for each installed package.
+* **DISTRO**:
+               Define the distro being used. Reference :ref:`ref-linux-distro`.
+* **TUF_TARGETS_EXPIRE**:
+               Is used to change the default target expiration date (default 1y).
+* **SSTATE_CACHE_MIRROR**:
+               It defaults to the directory mounted on the SDK build container
+               and if this directory exists it is used as the source for the
```
Comment (kprosise) [inline source/reference-manual/factory/factory-definition.rst:201]:
```suggestion
               Defaults to the directory mounted on the SDK build container.
               If this directory exists, it is used as the source for the
```

File: source/reference-manual/factory/factory-definition.rst [hunk 5]
```diff
@@ -169,11 +174,42 @@ lmp:
...
+               shared state cache (sstate-cache) mirror. When the directory does
+               not exist the ``lmp-manifest`` value is used and should point to
```
Comment (kprosise) [inline source/reference-manual/factory/factory-definition.rst:203]:
```suggestion
               not exist, the ``lmp-manifest`` value is used and should point to
```

**Review by kprosise — APPROVED** [review]
Some minor suggestions, mostly related to commas. LGTM!

File: source/reference-manual/factory/factory-definition.rst [hunk 6]
```diff
@@ -169,11 +174,42 @@ lmp:
...
+* **DISABLE_GPLV3**:
+               When set to ``1``, this variable configures the source code
+               to avoid the LmP default packages under GPLv3.
+               Reference: :ref:`ref-remove-gplv3`.
+* **ENABLE_PTEST**:
+               When set to ``1`` this variable creates an artifact based on the
+               image being built with the pattern ``<main tag>-ptest``. This
+               image includes the available Ptest for each installed package.
+* **DISTRO**:
+               Define the distro being used. Reference :ref:`ref-linux-distro`.
```
Comment (vanmaegima) [inline source/reference-manual/factory/factory-definition.rst:196]:
```suggestion
               Defines the distro being used. Reference :ref:`ref-linux-distro`.
```

File: source/reference-manual/factory/factory-definition.rst [hunk 7]
```diff
@@ -169,11 +174,42 @@ lmp:
...
+               shared state cache (sstate-cache) mirror. When the directory does
+               not exist the ``lmp-manifest`` value is used and should point to
+               the public HTTP shared state cache.
+
 .. _def-containers:
 
 containers
 ----------
 
+Define the container's configuration, including some image configuration and target architecture.
```
Comment (vanmaegima) [inline source/reference-manual/factory/factory-definition.rst:211]:
```suggestion
Defines the container's configuration, including some image configuration and target architecture.
```

**Review by vanmaegima — APPROVED** [review]

**Comment by doanac** [comment]
Docs for d8a14f4988983c839a376967cce43936870d3057 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2342/docs/artifacts/html/index.html

**Review by quaresmajose — APPROVED** [review]

File: source/reference-manual/factory/factory-definition.rst [hunk 8]
```diff
@@ -169,11 +174,42 @@ lmp:
...
+
 .. sidebar:: ``containers:`` Section Example
 
     .. code-block:: yaml
```
[the remark this answers is not in the document]
Comment (kprosise) [reply source/reference-manual/factory/factory-definition.rst:211]: You can if you want, or I can get to it when I do the cleanup of the reference manual.

[the remark this answers is not in the document]
Comment (angolini) [reply source/reference-manual/factory/factory-definition.rst:211]: I thought you had already cleaned those pages. I prefer to keep this for you, so. Thanks @kprosise

**Pull request merged**